### ISSUE TITLE
Lunar Release

### ISSUE THREAD

**Issue by mikaelarguedas** [ros-perception/perception_pcl#130]
Hi @paulbovbel and `perception_pcl` maintainers!

As mentioned in ros-perception/pcl_msgs#7, it would be great to have this repo released into Lunar asap to unblock the release of the navigation stack and all downstream packages. Please run bloom release on the current 1.4.1 version to release it into Lunar.

Thanks!

**Comment by paulbovbel** [comment]
Looks like PCL 1.8 in yakkety/zesty has split into multiple packages (http://packages.ubuntu.com/search?keywords=libpcl), where there is no longer a single debian like `libpcl1.7`.

Will have to investigate how CMakeLists and rosdep dependencies will change in lunar, given that we can no longer just `find_package(PCL 1.7 REQUIRED)`.

**Comment by mikaelarguedas** [comment]
hmm that's unconvenient. Thanks for looking into this!
If it's of any help, the `ros_buildfarm` prerelease jobs can help you test this on the new platforms (setting up containers and building the code). More info about running prerelease for repositories not part of the rosdistro index can be found [here](https://github.com/ros-infrastructure/ros_buildfarm/blob/master/doc/jobs/prerelease_jobs.rst#run-for-custom-repositories)

**Comment by wkentaro** [comment]
> Looks like PCL 1.8 in yakkety/zesty has split into multiple packages (http://packages.ubuntu.com/search?keywords=libpcl), where there is no longer a single debian like libpcl1.7.

I'm not sure, but `libpcl-dev` not work? Adding it to after [this line](https://github.com/ros/rosdistro/blob/master/rosdep/base.yaml#L2196)?

And should we release pcl_conversions first, right?

**Comment by mikaelarguedas** [comment]
I think that @paulbovbel is referring to how to find the package via CMake code and not the global rosdep key. The package `libpcl-dev` contains the package libpcl1.7 [on xenial](http://packages.ubuntu.com/xenial/libpcl-dev) so you can `find_package(PCL 1.7 REQUIRED)`.
But in [yakkety and later](http://packages.ubuntu.com/yakkety/libpcl-dev) there is no package libpcl1.8 but multiple libpclFEATURE1.8 so you may need to find_package them independently.

I didnt try to run `find_package(PCL 1.8 REQUIRED)` though...

**Comment by wkentaro** [comment]
I see.

BTW, are there any tools to test this kind of thing (I mean prepare releases)? Do you use docker + rosdep + wstool + rosinstall_generator + catkin-tools?

**Comment by wkentaro** [comment]
I tested on docker

**Xenial**

- OK

**Yakkety**

- Build passed with 
  - https://github.com/ros/rosdistro/pull/14524

- Remaining Error: rosdep Key error

```
$ rosdep install --from-path . -r -y

ERROR: the following packages/stacks could not have their rosdep keys resolved
to system dependencies:
pcl_ros: Cannot locate rosdep definition for [pcl_conversions]
Continuing to install resolvable dependencies...
executing command [sudo -H apt-get install -y libvtk-java]
Reading package lists... Done
Building dependency tree
Reading state information... Done
Package libvtk-java is not available, but is referred to by another package.
This may mean that the package is missing, has been obsoleted, or
is only available from another source

E: Package 'libvtk-java' has no installation candidate
ERROR: the following rosdeps failed to install
  apt: command [sudo -H apt-get install -y libvtk-java] failed
  apt: Failed to detect successful installation of [libvtk-java]
```


**Zesty**

- Build passed with 
  - https://github.com/ros/rosdistro/pull/14524
  - https://github.com/ros-perception/perception_pcl/pull/133

- Remaining Error: rosdep Key error: same as that on yakkety

**Comment by mikaelarguedas** [comment]
@wkentaro it looks like you figured out ways to test it. I'm answering here anyway because that can be useful for future releases. Usually I test things using the prerelease scripts provided by `python3-ros-buildfarm`.
If your package has a source entry on rosdistro you can use [prerelease.ros.org](http://prerelease.ros.org/)
Otherwise you can use the prerelease scripts on custom repositories (not indexed in rosdistro) folowing these instructions https://github.com/ros-infrastructure/ros_buildfarm/blob/master/doc/jobs/prerelease_jobs.rst#run-for-custom-repositories

**Comment by wkentaro** [comment]
Oh, this should be what I was looking for. Thanks!

**Comment by paulbovbel** [comment]
Please branch lunar-devel, and remove pointcloud_to_laserscan from it before blooming. That package has been here (https://github.com/ros-perception/pointcloud_to_laserscan) for some time.

**Comment by paulbovbel** [comment]
Also don't forget to add yourself as a maintainer to all the packages :) otherwise you won't get buildfarm emails or open source karma

**Comment by wkentaro** [comment]
Sent PR to ros/rosdistro: https://github.com/ros/rosdistro/pull/14672

**Comment by mikaelarguedas** [comment]
The initial builds were failing because of a wrong rosdep key (https://github.com/ros/rosdistro/pull/14676)
Now that https://github.com/ros/rosdistro/pull/14676 has been merged, can you please rerun bloom to trigger packages with the right key?
Thanks!

**Comment by wkentaro** [comment]
https://github.com/ros/rosdistro/pull/14677

**Comment by mikaelarguedas** [comment]
Looks like all jobs turned green, I'm going to close this. Thanks @wkentaro for the release!

**Comment by paulbovbel** [comment]
Thanks @wkentaro!
